### ISSUE TITLE
【Feature Request】Extending IngressRouteTCP for multiple entrypoints support

### ISSUE THREAD

**Issue by xzycn** [traefik/traefik#9517]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

Hi, I have been using traefik as a TCP LB for a while,recently I switch the configuration discovery from file to K8SCRD because I want a better network performance,so I need to jump over the IPVS's NAT(kube-proxy with IPVS mode),connect to Pod directly.
But now,there has another problem: in my case, I expose serveral thousands ports and then I need to create the same number of IngressRouteTCP object，which will cause a time-consuming problem when deploy with helm :)
So, can we put all route rules in one CRD object?

helm version: v3.9.0
k8s version: 1.24.4

**Comment by xzycn** [comment]
Reopen it please，it definitely is a **feature request**. 😅

**Comment by nmengin** [comment]
Hello @xzycn,

We've redirected you to the forum because, from our understanding, the feature you are requesting already exists.
Indeed, it's already possible de set many entrypoints to an `IngressRouteTCP` as described in [the documentation](https://doc.traefik.io/traefik/routing/providers/kubernetes-crd/#kind-ingressroutetcp).

But maybe we missed something, if so could you provide more information about your need, please?

**Comment by xzycn** [comment]
@nmengin 
Hi, I want the traffic to a port which you call entrypoint in traefik route to a specific bakend  service.
Indeed, now you can set many entrypoints to an  **IngressRouteTCP**, but cannot detach them with a route-match like **HostSNI** ,so if we have a route-match like **Port(*)**, it would be a solution.

**Comment by rtribotte** [comment]
Hello @xzycn,

We forgot to mention that any TCP router, that has no `entrypoints` defined, would be attached to all available entrypoints. Here is what the documentation says:
```
If not specified, TCP routers will accept requests from all defined entry points. 
If you want to limit the router scope to a set of entry points, set the entry points option.
```

Does this address your issue?

**Comment by rtribotte** [comment]
Hello @xzycn,

This is indeed much clearer!
Unfortunately, we have to decline this proposal.

To us, having such a matcher (a destination Port matcher) does not fit a real use case.
If a router is attached to an `entrypoint` it actually already matches connections on that port.
The port matcher would contravene the mechanism to attach a `router` to an `entrypoint`.

Moreover, one negative impact of this proposal is regarding resource consumption. Indeed, a lot of resources (router, middleware, service) would be built and never used.

Overall, we think that this proposal tends to provide configuration ease, but doesn’t fit with our design.
A valuable alternative to think about is to use an automation tool that would create the `IngressRouteTCP`resources.

**Comment by xzycn** [comment]
@rtribotte 
> A valuable alternative to think about is to use an automation tool that would create the IngressRouteTCPresources

Yes, I can do this through helm iteration. But this couldn't solve my problem mentioned above "**which will cause a time-consuming problem when deploy with helm**".
Another question, why dose it is a problem when we define all rules in one IngressRouteTCP, but seperating them in different 
IngressRouteTCP is acceptable? Sorry for that if I miss something :)

**Comment by mpl** [comment]
> @rtribotte
> 
> > A valuable alternative to think about is to use an automation tool that would create the IngressRouteTCPresources
> 
> Yes, I can do this through helm iteration. But this couldn't solve my problem mentioned above "**which will cause a time-consuming problem when deploy with helm**". Another question, why dose it is a problem when we define all rules in one IngressRouteTCP, but seperating them in different IngressRouteTCP is acceptable? Sorry for that if I miss something :)

Hello @xzycn ,

The underlying problem which makes the configuration model you proposed not acceptable, is the generation of useless/unused resources (routers, services, etc), which is indeed not directly visible at the configuration level.
In the example you proposed (port10000 - port10004), each matching rule would generate 5 routers (one router attached to each of the entrypoints, because a router gets generated for each entrypoint by default), and 4 of these routers would be useless, since they would have an incompatible port rule. So overall, you'd have 4*5 = 20 useless routers out of the 25 generated.

So yes, in terms of configuration it does not look like a problem, but behind the scenes it does not fit with how routers are matched with entrypoints in our model.

**Comment by xzycn** [comment]
@mpl well, thanks for your explanation , hope to see a alternative about new configuration model or another perfect way to handle my case in future. :)
